File: ansible-canary-webservice/ansible-web/com.redhat.ansible.web.parent/com.redhat.ansible.web.server2/src/main/java/com/redhat/ansible/web/server/model/AnsibleFacts.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.redhat.ansible.web.server.model;

/**
 *
 * @author dfreese
 */
public class AnsibleFacts extends CoreData {

    private Facts facts; 

    public Facts getFacts() {
        if(facts == null){
            facts = new Facts(); 
        }
        return facts;
    }

    public void setFacts(Facts facts) {
        this.facts = facts;
    }
    
}

File: ansible-canary-webservice/ansible-web-front/src/app/model/ansible-facts.ts
import { Facts } from './facts';

export class AnsibleFacts {
  hostname: string; 
  facts: Facts;

  constructor(values: Object = {}){
    Object.assign(this, values);
  }

  public getFacts(): Facts{
    return this.facts; 
  }

  public setFacts(facts: Facts){
      this.facts = facts; 
  } 
}